### PR TITLE
Mongo Replica set - Add replicaMemberConfig to replica instance

### LINKED ISSUE #209: slaveDelay option for MongoDB replicaSet
Currently, slaveDelay option is not supported in replica Set initialization.

https://docs.mongodb.com/manual/tutorial/configure-a-delayed-replica-set-member/

### PR DESCRIPTION
**Pull request by nhuanhoangduc**
## Replica Set Configuration - Add member config
Ref: https://docs.mongodb.com/manual/reference/replica-configuration/#replica-set-configuration-document-example

Sample code:
```javascript
const replSet = new MongoMemoryReplSet({
  instanceOpts: [
    {
      args: ["--quiet"],
      replicaMemberConfig: {
        arbiterOnly: true,
      },
    },
  ],
});

```

## Related Issues

- fixes #209

### CODE REVIEW

**Review by hasezoey — CHANGES_REQUESTED** [review]
looking good, just one thing

File: packages/mongodb-memory-server-core/src/util/MongoInstance.ts [hunk 1]
```diff
@@ -16,11 +16,23 @@ const log = debug('MongoMS:MongoInstance');
 
 export type StorageEngine = 'devnull' | 'ephemeralForTest' | 'mmapv1' | 'wiredTiger';
 
+// Ref: https://docs.mongodb.com/manual/reference/replica-configuration/#replica-set-configuration-document-example
```
[the remark this answers is not in the document]
Comment (nhuanhoangduc) [reply packages/mongodb-memory-server-core/src/util/MongoInstance.ts:19]: I will update tsdoc in detail

[the remark this answers is not in the document]
Comment (nhuanhoangduc) [reply packages/mongodb-memory-server-core/src/util/MongoInstance.ts]: Hi @hasezoey, I have updated tsdoc and unit test for this PR, please give me your feedbacks then I will improve this one. Thanks

**Comment by hasezoey** [comment]
btw, does this PR have something to do with #209 ?

**Review by hasezoey — APPROVED** [review]
thanks for add this, code and tests look good, but i will wait to merge until https://github.com/nodkz/mongodb-memory-server/pull/508#issuecomment-882395863 is answered

**Comment by nhuanhoangduc** [comment]
> btw, does this PR have something to do with #209 ?

I have checked mongo document version 4.4 and 5.0, there is a change in configuration to delay a replica set member:
For version 4.4: set options.slaveDelay = number
For version 5.x: set options.secondaryDelaySecs = number
My thought is we need to support both version (4.x and 5.x) since Mongodb 5.0 has ben release some days ago. I am going to add `slaveDelay` option to handle version 4.x

File: packages/mongodb-memory-server-core/src/util/MongoInstance.ts [hunk 2]
```diff
@@ -16,11 +16,85 @@ const log = debug('MongoMS:MongoInstance');
...
+   * @example
+   * ```ts
+   * { "<tag1>": "<string1>", "<tag2>": "<string2>",... }
+   * ```
+   */
+  tags?: any;
+
+  /**
+   * Mongodb 4.x only - The number of seconds "behind" the primary that this replica set member should "lag".
+   * @deprecated For mongodb 5.x, use `secondaryDelaySecs` instead.
```
Comment (hasezoey) [inline packages/mongodb-memory-server-core/src/util/MongoInstance.ts:72]: could this be removed? at least for as long as 4.0.x is the default for this package

**Review by hasezoey — CHANGES_REQUESTED** [review]

**Review by hasezoey — APPROVED** [review]

**Pull request merged**